apm-agent-core/src/main/java/co/elastic/apm/agent/metrics/builtin/SystemMetrics.java

```diff
@@ -78,26 +97,133 @@
+                    if (cgroupFilesTest != null) return cgroupFilesTest;
+
+                }
+            } catch (Exception e) {
+            }
+        }
+        return null;
+    }
+    String applyCgroup1Regex(String mountLine) {
+        Matcher matcher = CGROUP1_MOUNT_POINT.matcher(mountLine);
+        if (matcher.matches()) {
+            return matcher.group(1);
+        }
+        return null;
+    }
+    String applyCgroup2Regex(String mountLine) {
+        Matcher matcher = CGROUP2_MOUNT_POINT.matcher(mountLine);
+        if (matcher.matches()) {
+            return matcher.group(1);
+        }
+        return null;
+    }
+    private CgroupFiles verifyCgroup2Available(String lineCgroup, File mountDiscovered) throws IOException {
+        final String[] cgroupSplit = StringUtils.split(lineCgroup, ':');
+        // Checking cgroup2
+        File maxMemory = new File(mountDiscovered, cgroupSplit[cgroupSplit.length - 1] + "/" + CGROUP2_MAX_MEMORY);
+        if (maxMemory.canRead()) {
+            try(BufferedReader fileReaderMem = new BufferedReader(new FileReader(maxMemory))) {
+                String memMaxLine = fileReaderMem.readLine();
+                if (!"max".equalsIgnoreCase(memMaxLine)) {
+                    return new CgroupFiles(maxMemory,
+                        new File(mountDiscovered, cgroupSplit[cgroupSplit.length - 1] + "/" + CGROUP2_USED_MEMORY),
+                        new File(mountDiscovered, cgroupSplit[cgroupSplit.length - 1] + "/" + CGROUP2_STAT_MEMORY));
+                }
+            }
+        }
+        return null;
+    }
+
+    private CgroupFiles verifyCgroup1Available(File mountDiscovered) throws IOException {
+        // Checking cgroup1
+        File maxMemory = new File(mountDiscovered, CGROUP1_MAX_MEMORY);
+        if (maxMemory.canRead()) {
+            try(BufferedReader fileReaderMem = new BufferedReader(new FileReader(maxMemory))) {
+                String memMaxLine = fileReaderMem.readLine();
+                long memMax = Long.parseLong(memMaxLine);
+                if (memMax < UNLIMITED) { // Cgroup1 use a contant to disabled limits
+                    return new CgroupFiles(maxMemory,
+                        new File(mountDiscovered, CGROUP1_USED_MEMORY),
+                        new File(mountDiscovered, CGROUP1_STAT_MEMORY));
+                }
+            }
+        }
+        return null;
     }
 
     @Override
     public void start(ElasticApmTracer tracer) {
         bindTo(tracer.getMetricRegistry());
     }
 
+    private double getInactiveMemory() {
+        try(BufferedReader fileReaderStatFile = new BufferedReader(new FileReader(cgroupFiles.getStatMemory()))) {
+            long sum = 0;
+            for (String statLine = fileReaderStatFile.readLine(); statLine != null && !statLine.isEmpty(); statLine = fileReaderStatFile.readLine()) {
+                if (WildcardMatcher.isAnyMatch(inactiveMemoryRelevantLines, statLine)) {
```

Comment:
Let's not use a wildcard matcher here, search for exact line instead. Split each line and look at the first part for both `inactive_file` and `total_inactive_file`, preferring `total_inactive_file` if exists.